src/Core/Localization/DataLayer/CurrencyDataLayerInterface.php

```diff
@@ -70,5 +70,5 @@ public function write($currencyCode, CurrencyData $currencyData);
      *
      * @return self
      */
-    public function setLowerLayer(CurrencyDataLayerInterface $lowerLayer);
+    public function setLowerLayer(self $lowerLayer);
```

Comment:
As this is an interface I think it's better to keep the explicit Interface in the prototype